fix: content libraries permissions  [FC-0062]

rpenido opened the pull request:
## Description

This PR changes the permissions for content libraries so that only people who can create courses should be allowed to create new content libraries.

## Additional Information
- Related to https://github.com/openedx/edx-platform/issues/35943

## Testing instructions
- Read the code and check if the unit tests cover the expected behavior
- Go through the testing instructions from https://github.com/openedx/frontend-app-authoring/pull/1545

___
Private ref: [FAL-3987](https://tasks.opencraft.com/browse/FAL-3987)

openedx-webhooks commented:
Thanks for the pull request, @rpenido!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

rpenido commented:
Hi @bradenmacdonald and @ormsbee!
~This is ready for review!~

I had one doubt while manually testing it:

This code handles the visibility of the `+ New Library` button on the Studio, for both `library_v1` and `library_v2`:
https://github.com/openedx/edx-platform/blob/70b60ff256b32c5f5d0590a058f75533b778e2eb/cms/djangoapps/contentstore/views/library.py#L55-L70

Just to confirm, are Organization Staff, Course Staff and Course Admin also `course_creators` in this context?
I hadn't considered them yet.

Also, `ENABLE_CREATOR_GROUP=False` previously was "everyone can create libraries", and we are moving to "only global staff can create libraries". Is this intended? Are we diverging from `library_v1` permissions here?

bradenmacdonald commented:
@ormsbee or @jmakowski1123  can correct me if I'm wrong, but I don't think we want any org-specific permissions for now. It will be simply what's described on the ticket:

> When ENABLE_CREATOR_GROUP is False, only global staff may create new content libraries.
> When ENABLE_CREATOR_GROUP is True, people in the course creator group can create content libraries, along with global staff. Other users may not create libraries.

> Are we diverging from library_v1 permissions here?

Yes, we are diverging from library v1 permissions.

> This code handles the visibility of the + New Library button on the Studio, for both library_v1 and library_v2:

Let me look into this a bit more. Ideally for v2 the button should match the new behavior that you've added here.

rpenido commented:
> Let me look into this a bit more. Ideally for v2 the button should match the new behavior that you've added here.

No big issues. We can create another attribute to handle the `library_v2` case.

bradenmacdonald commented:
@rpenido OK, please do that.

rpenido commented:
> @rpenido OK, please do that.

@bradenmacdonald 
Here: https://github.com/openedx/frontend-app-authoring/pull/1545

rpenido commented:
This is ready for review @bradenmacdonald!

ormsbee commented on cms/djangoapps/contentstore/utils.py:
```diff
@@ -1565,6 +1568,7 @@ def get_library_context(request, request_is_json=False):
             'courses': [],
             'libraries_enabled': libraries_v1_enabled(),
             'show_new_library_button': user_can_view_create_library_button(request.user) and request.user.is_active,
+            'show_new_library_v2_button': user_can_create_library(request.user) and request.user.is_active,
```
Question: Isn't the `is_active` check redundant here?

rpenido replied:
Yes! Good catch!
Removed here: 30c6d06ef562f0790a08bf38fe3bb82ebaabaf58

ormsbee reviewed (approved):
Please squash your commits and add context to your commit message from the PR description. Otherwise, good to merge and create a backport PR into the sumac branch. Thank you!

ormsbee commented:
@rpenido: You'll need to rebase off the latest master to fix the static-assets-check check.

rpenido committed:
fix: content libraries permissions

This PR changes the permissions for content libraries so that only
people who can create courses should be allowed to create new content
libraries.

rpenido commented:
@ormsbee  Done!

Backport PR here https://github.com/openedx/edx-platform/pull/35979
